tests/simulator/test_advance.py

```diff
@@ -237,6 +240,94 @@ def test_field_coarsening_via_subcycles(self, refinement_boxes):
+            to the level field ghost of L1 of L0L1_datahier
+
+          The simulations are no longer comparable after the first advance, so this test cannot work beyond that.
+        """
+
+        print("test_field_coarsening_via_subcycles for dim/interp : {}/{}".format(ndim, interp_order))
+
+        from tests.amr.data.field.refine.test_refine_field import refine_time_interpolate
+        from pyphare.pharein import global_vars
+
+        import random
+        rando = random.randint(0, 1e10)
+
+        def _getHier(diag_dir, boxes=[]):
+            return self.getHierarchy(interp_order, boxes, "eb", cells=30,
+                time_step_nbr=1, smallest_patch_size=5, largest_patch_size=30,
+                diag_outputs=diag_dir, extra_diag_options={"fine_dump_lvl_max": 10}, time_step=0.001,
+                model_init={"seed": rando}
+            )
+
+        def assert_time_in_hier(*ts):
+            for t in ts:
+                self.assertIn(L0L1_datahier.format_timestamp(t), L0L1_datahier.times())
+
+        L0_datahier = _getHier(f"phare_lvl_ghost_interpolation_L0_diags_{self.ddt_test_id()}")
+        L0L1_datahier = _getHier(
+          f"phare_lvl_ghost_interpolation_L0L1_diags_{self.ddt_test_id()}", refinement_boxes
+        )
+
+        lvl_steps = global_vars.sim.level_time_steps
+        assert len(lvl_steps) == 2  # this test is only configured for L0 -> L1 refinement comparisons
+
+        coarse_ilvl = 0
+        fine_ilvl   = 1
+        coarsest_time_before = 0 # init
+        coarsest_time_after = coarsest_time_before + lvl_steps[coarse_ilvl]
+        assert_time_in_hier(coarsest_time_before, coarsest_time_after)
+
+        fine_subcycle_times = []
+        for fine_subcycle in range(global_vars.sim.level_step_nbr[fine_ilvl] + 1):
+            fine_subcycle_time   = coarsest_time_before + (lvl_steps[fine_ilvl] * fine_subcycle)
+            assert_time_in_hier(fine_subcycle_time)
+            fine_subcycle_times += [fine_subcycle_time]
+
+        quantities = [f"{EM}{xyz}" for EM in ["E", "B"] for xyz in ["x", "y", "z"]]
+        interpolated_fields = refine_time_interpolate(
+          L0_datahier, quantities, coarse_ilvl, coarsest_time_before, coarsest_time_after, fine_subcycle_times
+        )
+
+        checks = 0
+        for fine_subcycle_time in fine_subcycle_times:
+            fine_level_qty_ghost_boxes = level_ghost_boxes(L0L1_datahier, quantities, fine_ilvl, fine_subcycle_time)
+            for qty in quantities:
+                for fine_level_ghost_box_data in fine_level_qty_ghost_boxes[qty]:
+                    fine_subcycle_pd = fine_level_ghost_box_data["pdata"]
+                    for fine_level_ghost_box in fine_level_ghost_box_data["boxes"]:
+                        for refinedInterpolatedField in interpolated_fields[qty][fine_subcycle_time]:
+                            lvlOverlap = refinedInterpolatedField.box * fine_level_ghost_box
+                            if lvlOverlap is not None:
```

Comment:
can it be None? if not should we assert if it is? A priori if a box is a level ghost it *must* overlap the refined L0, no?

Reply:
it can be none if there's multiple L0 patches, when running under MPI for example

Reply:
right....